Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.IntegrationTests/ApiGatewayResponseExtensionsJsonInference.cs

```diff
@@ -0,0 +1,101 @@
+            {
+                Body = "Hello from lambda"
+            };
+
+            var httpContext = new DefaultHttpContext();
+            testResponse.ToHttpResponse(httpContext);
+            var actualResponse = await _httpClient.PostAsync(_fixture.JsonInferenceHttpApiUrl, new StringContent("Hello from lambda"));
+
+            await _fixture.ApiGatewayTestHelper.AssertResponsesEqual(actualResponse, httpContext.Response);
+            Assert.Equal(200, (int)actualResponse.StatusCode);
+            Assert.Equal("application/json", actualResponse.Content.Headers.ContentType.ToString());
+            var content = await actualResponse.Content.ReadAsStringAsync();
+            Assert.Equal("Hello from lambda", content);
+        }
+
+        [Fact]
+        public async Task V2_TestCanInferJsonType2()
+        {
+            var testResponse = new APIGatewayHttpApiV2ProxyResponse
+            {
+                Body = "{\"key\" : \"value\"}"
+            };
+
+            var httpContext = new DefaultHttpContext();
+            testResponse.ToHttpResponse(httpContext);
+            var actualResponse = await _httpClient.PostAsync(_fixture.JsonInferenceHttpApiUrl, new StringContent("{\"key\" : \"value\"}"));
+
+            await _fixture.ApiGatewayTestHelper.AssertResponsesEqual(actualResponse, httpContext.Response);
+            Assert.Equal(200, (int)actualResponse.StatusCode);
+            Assert.Equal("application/json", actualResponse.Content.Headers.ContentType.ToString());
+            var content = await actualResponse.Content.ReadAsStringAsync();
+            Assert.Equal("{\"key\" : \"value\"}", content);
+        }
+
+        [Fact]
+        public async Task V2_TestJsonInfersAlwaysUsesApplicationJson()
+        {
+            var payload = "{\"key\" : \"value\", \"headers\" : {\"content-type\": \"application/xml\"}}";
+
+            var testResponse = new APIGatewayHttpApiV2ProxyResponse
+            {
+                Body = payload
+            };
+
+            var httpContext = new DefaultHttpContext();
+            testResponse.ToHttpResponse(httpContext);
+
+            var actualResponse = await _httpClient.PostAsync(_fixture.JsonInferenceHttpApiUrl, new StringContent(payload));
+
+            await _fixture.ApiGatewayTestHelper.AssertResponsesEqual(actualResponse, httpContext.Response);
+            Assert.Equal(200, (int)actualResponse.StatusCode);
+            Assert.Equal("application/json", actualResponse.Content.Headers.ContentType.ToString());
+            var content = await actualResponse.Content.ReadAsStringAsync();
+
+            var responsePayload = JsonSerializer.Deserialize<Dictionary<string, object>>(content);
+            Assert.Equal("value", responsePayload["key"].ToString());
+        }
+
+        [Fact]
+        public async Task V2_HandlesNonJsonResponse()
+        {
+            var testResponse = new APIGatewayHttpApiV2ProxyResponse
```

Comment:
this case needs to be reviewed because its currently failing.it seems lambda functions always return valid json